connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/expressions/Aggregator.scala

```diff
@@ -52,7 +56,7 @@ import org.apache.spark.sql.{Encoder, TypedColumn}
  * @since 4.0.0
  */
 @SerialVersionUID(2093413866369130093L)
-abstract class Aggregator[-IN, BUF, OUT] extends Serializable {
+abstract class Aggregator[-IN: TypeTag, BUF, OUT] extends Serializable {
```

Comment:
There are a few concerns here:
- How does this work with java implementations? How ugly is this going to be.
- This is a breaking change compared to the sql/core class. I am fine adding this. However I'd probably homogenise the implementations.

Reply:
I reverted this change and switched to use reflection. The new method is hacky but should work in all cases.